source/MRMesh/MRIRenderObject.h

```diff
@@ -44,6 +48,38 @@ struct ModelRenderParams : BaseRenderParams
     bool alphaSort{ false };     // if this flag is true shader for alpha sorting is used, unused for picker
 };
 
+struct UiRenderParams
+{
+    /// Multiply all your sizes by this amount. Unless they are already premultipled, e.g. come from `ImGui::GetStyle()`.
+    float scale = 1;
+
+    /// The current viewport.
+    const Viewport* viewport = nullptr;
+
+    /// For convenience, the viewport corner coordinates (with Y going down).
+    Vector2f viewportCornerA;
+    Vector2f viewportCornerB;
```

Comment:
ViewportId viewportId;       // id of the viewport
    Vector4i viewport;           // viewport x0, y0, width, height

like in `BaseRenderParams`

Reply:
Inherited from `BaseRenderParams` instead.